iOS/Projects/Features/SignUp/Sources/Presentation/SignUpProfileScene/ViewModel/SignUpProfileViewModel.swift

```diff
@@ -0,0 +1,167 @@
+//
+//  SignUpProfileViewModel.swift
+//  SignUpFeature
+//
+//  Created by 안종표 on 12/6/23.
+//  Copyright © 2023 kr.codesquad.boostcamp8. All rights reserved.
+//
+
+import Combine
+import Foundation
+import Log
+
+// MARK: - SignUpProfileViewModelInput
+
+public struct SignUpProfileViewModelInput {
+  let nickNameTextFieldEditting: AnyPublisher<String, Never>
+  let imageButtonTap: AnyPublisher<Void, Never>
+  let imageSetting: AnyPublisher<Data, Never>
+  let completeButtonTap: AnyPublisher<Void, Never>
+  let genderBirth: AnyPublisher<GenderBirth, Never>
+}
+
+public typealias SignUpProfileViewModelOutput = AnyPublisher<SignUpProfileState, Never>
+
+// MARK: - SignUpProfileState
+
+public enum SignUpProfileState {
+  case idle
+  case checking(Bool)
+  case customError(Error)
+  case imageButtonTap
+  case image(Data)
+  case success
+  case failure
+}
+
+// MARK: - SignUpProfileViewModel
+
+public final class SignUpProfileViewModel {
+  private var subscriptions: Set<AnyCancellable> = []
+  private let nickNameCheckUseCase: NickNameCheckUseCaseRepresentable
+  private let imageTransmitUseCase: ImageTransmitUseCaseRepresentable
+  private let signUpUseCase: SignUpUseCaseRepresentable
+
+  private let userBit: UserBit
+
+  public init(
+    nickNameCheckUseCase: NickNameCheckUseCaseRepresentable,
+    imageTransmitUseCase: ImageTransmitUseCaseRepresentable,
+    signUpUseCase: SignUpUseCaseRepresentable,
+    userBit: UserBit
+  ) {
+    self.nickNameCheckUseCase = nickNameCheckUseCase
+    self.imageTransmitUseCase = imageTransmitUseCase
+    self.signUpUseCase = signUpUseCase
+    self.userBit = userBit
+  }
+}
+
+// MARK: SignUpProfileViewModelRepresentable
+
+extension SignUpProfileViewModel: SignUpProfileViewModelRepresentable {
+  public func transform(input: SignUpProfileViewModelInput) -> SignUpProfileViewModelOutput {
+    for subscription in subscriptions {
+      subscription.cancel()
+    }
+    subscriptions.removeAll()
+
+    var imageData: Data?
+
+    let imageFormSubject = PassthroughSubject<ImageForm, Never>()
+    let nickNameSubject = PassthroughSubject<String, Never>()
+
+    let nickNameCheckedResult = input.nickNameTextFieldEditting
+      .tryMap { [weak self] nickName in
+        guard let result = self?.nickNameCheckUseCase.check(nickName: nickName) else {
+          throw SignUpProfileViewModelError.invalidBinding
+        }
+        nickNameSubject.send(nickName)
+        return SignUpProfileState.checking(result)
+      }
+      .catch { Just(.customError($0)) }
+      .eraseToAnyPublisher()
+
+    let image = input.imageButtonTap
+      .flatMap { Just(SignUpProfileState.imageButtonTap) }
+      .eraseToAnyPublisher()
+
+    let imageSettingResult = input.imageSetting
+      .flatMap { Just(SignUpProfileState.image($0)) }
```

Comment:
p3: map으로 해도 충분할 것 같습니다.
```suggestion
      .map(SignUpProfileState.imageButtonTap)
      .eraseToAnyPublisher()

    let imageSettingResult = input.imageSetting
      .map(SignUpProfileState.image($0))
```